content/articles/how-to-handle-file-uploads-with-flask/index.md

```diff
@@ -0,0 +1,255 @@
+
+- Enabling file uploads
+
+### Reguirement
+In this guide, we will be using python 3, and VS Code text editor you can download [vscode](https://code.visualstudio.com/download) and [python](https://www.python.org/downloads/)
+
+### Python Virtual Environment
+A virtual environment is an isolated environment for python projects, this is a module created by python called venv, it gives a developer a unique environment that enables the installation of all packages that are unique to a particular project, the virtual environment doesn’t change the default python version or default packages installed in a system, instead, it gives you freedom from the interference of other packages installed in the system, this makes it easy to run any python project in any computer irrespective of python version or packages installed in the system.
+### How To Create a Virtual Environment  
+The process of creating a virtual environment differs base on the operating system in this guide we will look at the process in the context of a windows operating system. You can check the procedure online if you are using a different operating system.
+To start open PowerShell and make a directory using the command 
+```python   
+mkdir
+```
+cd into the new directory
+then install virtual environment using the command
+```
+pip install virtualenv
+```
+Then create the virtual environment using the command 
+```
+virtualenv myenv
+```
+Note that myenv is the name of my virtual environment it can be anything you wish
+Then activate the virtual environment using the command
+```python
+myenv/Scripts/activate
+```
+If you are using CMD your command will be 
+```
+myenv\Scripts\activate.bat
+``` 
+
+### Creating Our Project
+After activating our virtual environment we can now create our project, to do that we will make a new directory for the project. Use the below command 
+
+```
+mkdir tutorial
+```
+NOTE this is my project name choice yours can be anything.
+To build a flask application we must install flask, to do that we will use the below command 
+```
+pip install flask 
+
+```
+After installation, we will create a new file with the name ```app.py```, update the file with the below code
+
+```python
+from flask import Flask
+
+app = Flask(__name__)
+
+
+@app.route('/')
+def index():
+    return"hello world"
+
+if __name__==('__main__'):
+    app.run(debug=True)
+```
+from the code above we are importing flask from the flask library we installed.
+
+The ```@app.route``` is doing the routing for us, index is our view function which will return our page content to the browser.
+
+The if statement return the ```app.run``` which will enable us to run our app and refresh our page whenever we save changes.
+
+to run our app we run the below command on our terminal
+```python
+python app.py
+```
+note that ```app.py``` is the name of my app yours can be different. if everything goes well you will have a result like the one shown below.
+
+[image1]
```

Comment:
Please update all your image paths as I did below. Ensure you give accurate image names that relate to the image you want to show and meaningful alt tags

```suggestion
![App running](/engineering-education/how-to-handle-file-upload-in-flask/image1.PNG)
```